lisp/x/profiler/callgrind.go

```diff
@@ -195,11 +197,15 @@ func (p *callgrindProfiler) end(fun *lisp.LVal) {
 	memory := ref.endMemory - ref.startMemory
 	// Cache the location - we won't be able to get it again when we build the maps for
 	// things that call this.
-	if ref.line == 0 && fun.Source != nil {
-		ref.line = fun.Source.Line
-		ref.file = fun.Source.File
+	if ref.line == 0 && loc != nil {
+		ref.line = loc.Line
+		ref.file = loc.File
 	}
 	// Output timing and line ref
+	line := 0
+	if loc != nil {
+		line = loc.Line
```

Comment:
is it OK that `ref.line` may not be the same as this line here?

Reply:
I'm pretty sure this should work as it did before unless I missed something.